persistence/rest/src/main/java/org/infinispan/persistence/rest/RestStore.java

```diff
@@ -128,6 +125,7 @@ protected void initChannel(SocketChannel ch) {
       } catch (EncoderException e) {
       }
       this.metadataHelper = Util.getInstance(configuration.metadataHelper(), ctx.getCache().getAdvancedCache().getClassLoader());
+      new HttpResponseHandler(); // Initialize the Netty TypeParameterMatcher
```

Comment:
Can you add some more comments around this? This line has a really bad code smell. What happens if this line isn't here for example?